tests/src/python/test_provider_shapefile.py

```diff
@@ -629,7 +630,7 @@ def testOpenWithFilter(self):
             f = next(vl.getFeatures())
             self.assertEqual(f['name'], 'Apple')
             # force close of data provider
-            vl.setDataSource('', 'test', 'ogr')
+            vl.setDataSource('', 'test', 'ogr', options)
```

Comment:
Can we add this as a new test to be sure the old way still works

Reply:
The old way does not works, it is based on an unvirtual deprecated method only defined for QgsVectorLayer 88c0d3f07db541b1cb6d5cc81f4aabc12741de26

I can move this method to QgsMapLayer to keep API

Reply:
The old way is a deprecated way, do we need to keep test that use deprecated method ?